### PR TITLE
fixes #2536 - loading locale .mo files in production mode

### PR DESCRIPTION
**Pull request by tstrachota**
- initializer script fixed to look for *.mo files
- Makefile updated to place _.mo into /locale/_/LC_MESSAGES
  (rake gettext:pack places them there as well)
- *.mo files added to foreman.spec

### CODE REVIEW

File: foreman.spec [hunk 1]
```diff
@@ -351,7 +351,12 @@ plugins required for Foreman to work.
   sed -ri 'sX/usr/bin/rakeX%{scl_rake}X' extras/dbmigrate
 %endif
 
+#build locale files
+make -C locale all-mo
```
Comment (domcleal) [inline foreman.spec:355]: Should we prefer `rake locale:pack`, as that's our documented method?

Comment (lzap) [inline foreman.spec:355]: This is not documented, but I have put this on the translators wiki - this makefile offers extra checkers in the "check" target using python-based tool "pofilter". Also we can use it in the SPEC file to speed up our builds cos it is much faster. I would vote for keeping it here.

File: locale/Makefile [hunk 2]
```diff
@@ -6,13 +6,18 @@
...
+MOFILE = $(DOMAIN).mo
 POFILES = $(shell find . -name '*.po')
 MOFILES = $(patsubst %.po,%.mo,$(POFILES))
 POXFILES = $(patsubst %.po,%.pox,$(POFILES))
 
 %.mo: %.po
 	msgfmt -o $@ $<
+	mkdir -p $(patsubst %/$(MOFILE),%/LC_MESSAGES/,$@)
+	mv $@ $(patsubst %/$(MOFILE),%/LC_MESSAGES/$(MOFILE),$@)
+
```
Comment (lzap) [inline locale/Makefile]: But this is a hack :-) Why not to use

```
msgfmt -o ./LC_MESSAGES/$<
```

maybe you will need to provide full path, then you can use $(basedir $@) or something similar...

?

File: locale/Makefile [hunk 3]
```diff
@@ -53,3 +58,4 @@ tx-update: tx-pull update-po
 clean:
 	-rm -f messages.mo
 	find . \( -name "*.mo" -o -name "*.pox" \) -exec rm -f '{}' ';'
+	find . -path *LC_MESSAGES -exec rm -rf '{}' ';'
```
Comment (lzap) [inline locale/Makefile]: Why not to keep the empty directories there? I suggest to put LC_MESSAGES into the .gitignore then...

Anyway good work thanks for nice fix.

Comment (tstrachota) [inline locale/Makefile]: I look at it this way: Make created the directory, make should remove it.

File: foreman.spec (hunk 1 above)
Comment (tstrachota) [inline foreman.spec:355]: I prefer make as well and would keep it if you're not strongly against.

**Comment by domcleal** [comment]
Thanks @tstrachota!  Merged as 40b31d27c.

**Pull request closed without merging**